torchci/lib/drciUtils.ts

```diff
@@ -315,3 +315,18 @@ export async function hasSimilarFailures(
 
   return false;
 }
+
+export function isInfraFlakyJob(job: RecentWorkflowsData): boolean {
```

Comment:
Prob worth mentioning that this is not a documented github behavior and is subject to potentially changing (but we can take advantage of it while it lasts since it fails gracefully).

Reply:
Just FYI, there is actually another case of infra flaky that I still need to figure out how it can be classified correctly.  https://github.com/pytorch/pytorch/pull/110668 is an example where [one job](https://github.com/pytorch/pytorch/actions/runs/6425950591) fails because of the infamous

```
The self-hosted runner: i-0abf34dcfd6b840ab lost communication with the server. Verify the machine is running and has a healthy network connection. Anything in your workflow that terminates the runner process, starves it for CPU/Memory, or blocks its network access can cause this error.
```

This case is an infra flake, but there could be rare legit OOM build issues (remember the FlashAttentionV2 build) that end up with exactly the same error message.  So, it's hard to separate them.  Basically, if it's really OOM, it's legit failure.  Otherwise, it's flaky infra.